File: models/paper.py
from datetime import datetime
from extensions import db

class Paper(db.Model):
    __tablename__ = 'paper'
    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
    course = db.Column(db.String(100), nullable=False)
    year = db.Column(db.Integer, nullable=False)
    term = db.Column(db.String(20))
    college = db.Column(db.String(100))
    image_path = db.Column(db.String(255))
    created_at = db.Column(db.DateTime, default=datetime.utcnow) 

File: backend-go/models/paper.go
package models

import "time"

type Paper struct {
	ID          uint      `gorm:"primaryKey;autoIncrement" json:"id"`
	Course      string    `gorm:"type:varchar(100);not null" json:"course"`
	Year        int       `gorm:"not null" json:"year"`
	Term        string    `gorm:"type:varchar(20)" json:"term"`
	College     string    `gorm:"type:varchar(100)" json:"college"`
	ImagePath   string    `gorm:"type:varchar(255)" json:"image_path"`
	FileType    string    `gorm:"type:varchar(20);default:image" json:"file_type"`
	FileSize    int       `gorm:"default:0" json:"file_size"`
	StorageType string    `gorm:"type:varchar(20);default:local" json:"storage_type"`
	CreatedAt   time.Time `json:"created_at"`
}

func (Paper) TableName() string {
	return "paper"
} 